Lib/sysconfig.py

```diff
@@ -545,15 +545,32 @@ def _init_non_posix(vars):
     """Initialize the module as appropriate for NT"""
     # set basic install directories
     import _imp
+    import re
     vars['LIBDEST'] = get_path('stdlib')
     vars['BINLIBDEST'] = get_path('platstdlib')
     vars['INCLUDEPY'] = get_path('include')
-    try:
+
+    extension_suffixes = _imp.extension_suffixes()
+    if len(extension_suffixes) >= 1:
         # GH-99201: _imp.extension_suffixes may be empty when
         # HAVE_DYNAMIC_LOADING is not set. In this case, don't set EXT_SUFFIX.
-        vars['EXT_SUFFIX'] = _imp.extension_suffixes()[0]
-    except IndexError:
-        pass
+        # e.g., "_d.cp313-win_amd64.pyd"
+        vars['EXT_SUFFIX'] = extension_suffixes[0]
+
+        # e.g., "cp313-win_amd64"
+        _, soabi, _ = vars['EXT_SUFFIX'].split('.')
+        vars['SOABI'] = soabi
+
+        # e.g., check for a "t" (for "threading") in SOABI
+        if re.match(r'cp\d+t', soabi):
```

Comment:
The string processing is a bit icky. It would be convenient to just pass these values (`SOABI`, `Py_NOGIL`) from a C to Python, but it wasn't clear to me where to put that (i.e., in `sys`, `_imp`, `_winapi` or someplace else)

Reply:
I agree. Maybe a good option would be putting them on `_winapi` and then simply loading them in `_init_non_posix`?

Reply:
`sys.flags`?

Reply:
But this information (`SOABI`, `Py_NOGIL`, etc) doesn't seem quite right there, I think. Though maybe this is just my interpretation.

I have a prototype locally to export this information in `_winapi`, avoiding any public API changes. It doesn't feel amazing either, but :shrug:.

Really, this should just be exposed by sysconfig directly, which may involve a C `_sysconfig` module that exports this information out to Python, or something else. It's unfortunate that there's such a large discrepancy between how Makefile and Windows builds work :disappointed:. This is kind of a mess, and we should probably try to unify things, it's just... hard, and a lot of work. I'm a bit tired of all this 😕

Reply:
I moved some of these to `_winapi`. A `_sysconfig` C module might also make sense, but it would be very small.